osf/management/commands/sync_datacite_doi_metadata.py

```diff
@@ -10,31 +10,38 @@
 
 logger = logging.getLogger(__name__)
 
-def sync_datacite_doi_metadata(dry_run=True, retries=4):
+def retry(func, retries=4):
```

Comment:
Accepting an arbitrary func here seems needlessly generalized for a function outside of a utility module, especially considering you're assuming the args to `func`. Just make this explicitly something like `get_doi_with_retries`.

Reply:
ok I'm just going to move the logic out of that function into the main loop.